modules/gapi/samples/face_detection_mtcnn.cpp

```diff
@@ -364,7 +364,7 @@ GAPI_OCV_KERNEL(OCVR_O_NetPreProcGetROIs, R_O_NetPreProcGetROIs) {
         for (const auto& face : in_faces) {
             cv::Rect tmp_rect = face.bbox.getRect();
             //Compare to transposed sizes width<->height
-            tmp_rect &= cv::Rect(tmp_rect.x, tmp_rect.y, in_image_size.height - tmp_rect.x - 4, in_image_size.width - tmp_rect.y - 4);
+            tmp_rect &= cv::Rect(std::max(tmp_rect.x, 0), std::max(tmp_rect.y, 0), in_image_size.height - tmp_rect.x - 4, in_image_size.width - tmp_rect.y - 4);
```

Comment:
if `tmp_rect.x` is negative, then probably the `width` argument is not calculated properly too (e.g, it may exceed image size).

What is about this?
```
tmp_rect &= cv::Rect(...previous variant...) & cv::Rect(0, 0, in_image_size.height - 4, in_image_size.width - 4)
```

Reply:
+ if possible, please give a name for 4 :)

Reply:
+1

Reply:
I'm stuck trying to give a name for 4. So I just removed it. :)